agent/auto-config/auto_config.go

```diff
@@ -12,10 +12,10 @@ import (
 	"strings"
 	"time"
 
-	"github.com/hashicorp/consul/agent/agentpb"
 	"github.com/hashicorp/consul/agent/config"
 	"github.com/hashicorp/consul/lib"
 	"github.com/hashicorp/consul/logging"
+	pbautoconf "github.com/hashicorp/consul/proto/autoconf"
```

Comment:
Would it make it sense to actually name the package `github.com/hashicorp/consul/proto/pbautoconf` if we're always going to name the import?

This would also help disambiguate  `github.com/hashicorp/consul/proto/config` from `github.com/hashicorp/consul/agent/config` if it were named `pbconfig`.

Reply:
I wonder if there could be a middle ground of keeping the fs paths without the prefix but having the packages themselves have a prefix.

Something about a path like "github.com/hashicorp/consul/proto/pbautoconf" really irks me because proto is already in the import path.

Would this work better?

Reply:
The stdlib has several of these sorts of things though:
- `hash/maphash`
- `io/ioutil`
- `log/syslog`
- `net/http/httptest`
- `net/http/httptrace`
- `net/http/httputil`
- `net/rpc/jsonrpc`
- `path/filepath`
- `testing/iotest`

I feel like if you always have to import the package with a different name, then the package is misnamed.

Reply:
Fair enough. Although in this case it would be "occasionally" and not "always". Regardless I can add the prefixes.